### PR TITLE
Node API and example on how to use

### PR DESCRIPTION
**Pull request by jstroem**

### CODE REVIEW

File: node/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (rikkegammelmark) [inline node/package.json:1]: Any specific reason to use npm instead of yarn?

**Review by andersjohnsen — APPROVED** [review]

**Review by mmedum — APPROVED** [review]
LGTM

File: node/example/main.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const channel = new grpc.Channel("api.toit.io",
+    grpc.credentials.combineChannelCredentials(credentials,
+      grpc.credentials.createFromMetadataGenerator((_, cb) => {
+      const md = new grpc.Metadata();
+      md.set("Authorization", "Bearer " + new Buffer(auth.getAccessToken_asU8()).toString("utf8"));
+      cb(null, md);
+    })), {});
+
+  const devices = await listDevices(channel);
+  devices.forEach((d:Device) => {
```
Comment (rikkegammelmark) [inline node/example/main.ts:28]: Inconsistent spacing, I prefer "variblename: type"

File: node/example/main.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      console.log(d.getConfig().getName());
+  });
+}
+
+function listDevices(channel : grpc.Channel) : Promise<Array<Device>> {
+  return new Promise<Array<Device>>((resolve, reject) => {
+    const client = new DeviceServiceClient("", null, {channelOverride: channel});
+    const request = new ListDevicesRequest();
+    client.listDevices(request, function(err : Error, response : ListDevicesResponse) {
```
Comment (rikkegammelmark) [inline node/example/main.ts:37]: The type of the request callback is:
```
export type requestCallback<ResponseType> =
    (error: ServiceError | null, value?: ResponseType) => void;
```
So I don't think you can assume response is not undefined if err is null.

**Review by rikkegammelmark — APPROVED** [review]

File: node/package.json (hunk 1 above)
Comment (jstroem) [reply node/package.json:1]: removed lock file.

**Pull request merged**